src/include/btree_cmp_inline.h

```diff
@@ -116,8 +124,20 @@ __wt_compare(WT_SESSION_IMPL *session, WT_COLLATOR *collator, const WT_ITEM *use
  *     Check if the prefix item is equal to the leading bytes of the tree item.
  */
 static inline int
-__wt_prefix_match(const WT_ITEM *prefix, const WT_ITEM *tree_item)
+__wt_prefix_match(WT_SESSION_IMPL *session, const WT_ITEM *prefix, const WT_ITEM *tree_item)
 {
+    WT_BTREE *btree;
+    size_t len;
+
+    btree = S2BT(session);
+    len = strlen(btree->key_format);
+    /*
+     * Assert that we only perform prefix comparisons when the key format of the table is either a
+     * string or raw byte array.
+     */
+    WT_ASSERT(session,
+      WT_STREQ(btree->key_format, "S") || WT_STREQ(btree->key_format, "u") ||
```

Comment:
@keithbostic @luke-pearson Please review this assert. 

We were discussing on methods to check if a key format is valid fixed length string e.g. (5s, or 10s). The original idea was to check the last character in the `key_format`, and check if it is 's'. I have incorporated that change into this PR. However this method poses a problem when dealing with combination of formats, which we should be disallowing. For example u10s, will pass this assert when it actually should be fired. Do you think there is a better method we can manage around this? 

@etienneptl  and I thought about traversing the key format string, and manually check through each format inside the string, and when we see any other format we assert.

Reply:
Should we also be using the` btree->key_format `or pass in the WT_CURSOR_BTREE* cbt from the caller and get it from `cbt->iface.key_format`?

Reply:
@jiechenbo, I hope you don't mind me jumping in here.  It's good to have assertions about the `key_format`, but might also be good to have checks in `__wt_cursor_reconfigure` to make an equivalent check so that `"prefix_key"` must be used with "s", "S", "u", giving an `EINVAL` error.  That way, if API is misused, you get an error, rather than a crash.  And also worth having a python test that tests that error condition.

Reply:
@ddanderson Thanks! That's really good idea. I have moved the assert into `EINVAL` error instead.